src/main/java/com/onarandombox/MultiverseCore/utils/PermissionsTool.java

```diff
@@ -0,0 +1,329 @@
+package com.onarandombox.MultiverseCore.utils;
+
+import com.dumptruckman.minecraft.util.Logging;
+import com.onarandombox.MultiverseCore.MultiverseCore;
+import com.onarandombox.MultiverseCore.api.Destination;
+import com.onarandombox.MultiverseCore.api.MVWorld;
+import com.onarandombox.MultiverseCore.destination.ParsedDestination;
+import org.bukkit.command.CommandSender;
+import org.bukkit.permissions.Permission;
+import org.bukkit.permissions.PermissionDefault;
+import org.bukkit.plugin.PluginManager;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A utility class for registering and checking permissions.
+ */
+public class PermissionsTool {
+    private static final String TELEPORT_PERM_PREFIX = "multiverse.teleport.";
+
+    private final MultiverseCore plugin;
+    private PluginManager pluginManager;
+    private Permission worldAccessPermission;
+    private Permission bypassGameModePermission;
+    private Permission bypassEntryFeePermission;
+    private Permission bypassPlayerLimitPermission;
+
+    public PermissionsTool(MultiverseCore plugin) {
+        this.plugin = plugin;
+    }
+
+    /**
+     * Sets up the permissions for Multiverse.
+     */
+    public void setUpPermissions() {
```

Comment:
I wonder if this should also be `registerPermissions` since it's registering permissions just like some of the other public methods in this class. Maybe the plugin manager bit belongs elsewhere? Or `setUpWorldPermissionsWildcards` could be public `registerWildcardWorldPermissions()`?